### DISCUSSION TITLE
How can I contribute to this project?

### DISCUSSION THREAD

**Discussion by teggotic** [mint-lang/mint#174] [Answered] [answered]
I love idea of this language, but I don't know how to start contributing

**Comment by gdotdesign** [comment, marked as answer]
Hey there @Danil-Lapirow,

There are a number ways you can contribute:
- **use the language** - this is the most helpful thing at this stage because we can discover bugs and missing features this way
- **documentation and website** - the documentation is always needs some work, if you discover that something is not documented or can be improved you can create a PR for it in the websites repository (https://github.com/mint-lang/mint-website-rails)
- **language features** - if you have any idea about a new language feature, create a detailed issue about it with examples and description (why is it needed? what problems does it solve?)
- **code review** - the compiler can use a thorough code review because I don't write perfect code, also code reviews for PRs are welcome (@Sija always helps with this :+1:)
- **standard library** - the standard library is incomplete and needs a lot of work: 
  - create modules for not implemented Web APIs (or a separate package)
  - `Time` module needs some rethinking and possibly a rewrite
  - a lot of modules like `String`, `Dom`, etc... is missing a lot of features,  you can add new functions here (with tests)
- **write a package** - if you have a feature you use and can be moved into a package it can be good for other developers
- **marketing** - write blog posts and such to make other people aware of the language
- **compiler** - there are always a few issues that can be implemented in the compiler

If I can think of something else, I'll update this list.

Let me know if have any questions, we can discuss here or in Gitter (https://gitter.im/mint-lang/Lobby)

**Comment by teggotic** [comment]
Thank you. It'll be a good idea to post this somewhere. @gdotdesign

**Comment by s0kil** [comment]
Yes, It will be a good addition somewhere towards the bottom of the README.

**Comment by gdotdesign** [comment]
I agree that is should be in the README (merged the PR thanks @s0kil) also pinned this issue.

**Comment by Namek** [comment]
@Danil-Lapirow  Some deterministic (to work in a pure functional manner) `Random` module would be nice to have. Basically seed generation (not pure) and number generation based on the seed/previous seed.
